sdk/core/azure-core/tests/test_base_polling.py

```diff
@@ -347,6 +351,8 @@ def mock_send(http_request, http_response, method, status, headers=None, body=RE
             request,
             response,
         )
+        if is_rest(response):
+            response._body()
```

Comment:
Because we were using the backward compat layer of Rest response (`.body()`), which was hacking some internal thing to pre-load the body in the sync implementation. The current implementation is not using the backward compat layer, and calling directly `content`. This also means that now, in order for LRO to work, the content is required to be pre-loaded in memory. This is always the case, as no LRO code can be generated by passing `stream=False`. Since the test was assuming that the body will be laze loaded by the backward compat layer, when moving to `content` it was failing.
Note that since async pre-load is async, async LRO was already assuming that the body was preloaded. We forgot to consider this in the sync implementation.